remote-api/src/test/java/org/alfresco/rest/api/impl/CategoriesImplTest.java

```diff
@@ -197,5 +214,165 @@ public void testGetCategoryById_nodeNotExists()
+        then(categoryServiceMock).should().getRootCategoryNodeRef(StoreRef.STORE_REF_WORKSPACE_SPACESSTORE);
+        then(categoryServiceMock).should().createCategory(parentCategoryNodeRef, CATEGORY_NAME);
+        then(categoryServiceMock).shouldHaveNoMoreInteractions();
+
+        assertEquals(1, createdCategories.size());
+        Category createdCategory = createdCategories.iterator().next();
+        assertEquals(CATEGORY_ID, createdCategory.getId());
+        assertEquals(CATEGORY_NAME, createdCategory.getName());
+        assertEquals(PATH_ROOT, createdCategory.getParentId());
+        assertFalse(createdCategory.getHasChildren());
+    }
+
+    @Test
+    public void testCreateCategory()
+    {
+        given(authorityServiceMock.hasAdminAuthority()).willReturn(true);
+        final NodeRef parentCategoryNodeRef = new NodeRef(StoreRef.STORE_REF_WORKSPACE_SPACESSTORE, PARENT_ID);
+        given(nodesMock.validateNode(PARENT_ID)).willReturn(parentCategoryNodeRef);
+        given(nodesMock.isSubClass(parentCategoryNodeRef, ContentModel.TYPE_CATEGORY, false)).willReturn(true);
+        final NodeRef categoryNodeRef = prepareCategoryNodeRef();
+        given(categoryServiceMock.createCategory(parentCategoryNodeRef, CATEGORY_NAME)).willReturn(categoryNodeRef);
+        given(nodesMock.getNode(CATEGORY_ID)).willReturn(prepareCategoryNode());
+        final ChildAssociationRef parentAssoc = new ChildAssociationRef(null, parentCategoryNodeRef, null, categoryNodeRef);
+        given(nodeServiceMock.getPrimaryParent(categoryNodeRef)).willReturn(parentAssoc);
+        given(nodeServiceMock.getParentAssocs(parentCategoryNodeRef)).willReturn(List.of(parentAssoc));
+        given(nodeServiceMock.getChildAssocs(categoryNodeRef, RegexQNamePattern.MATCH_ALL, RegexQNamePattern.MATCH_ALL, false))
+                .willReturn(Collections.emptyList());
+
+        //when
+        final List<Category> createdCategories = objectUnderTest.createSubcategories(PARENT_ID, prepareCategories(), parametersMock);
+
+        then(authorityServiceMock).should().hasAdminAuthority();
+        then(authorityServiceMock).shouldHaveNoMoreInteractions();
+        then(nodesMock).should().validateNode(PARENT_ID);
+        then(nodesMock).should().isSubClass(parentCategoryNodeRef, ContentModel.TYPE_CATEGORY, false);
+        then(nodesMock).should().getNode(CATEGORY_ID);
+        then(nodesMock).shouldHaveNoMoreInteractions();
+        then(nodeServiceMock).should().getPrimaryParent(categoryNodeRef);
+        then(nodeServiceMock).should().getParentAssocs(parentCategoryNodeRef);
+        then(nodeServiceMock).should().getChildAssocs(categoryNodeRef, RegexQNamePattern.MATCH_ALL, RegexQNamePattern.MATCH_ALL, false);
+        then(nodeServiceMock).shouldHaveNoMoreInteractions();
+        then(categoryServiceMock).should().createCategory(parentCategoryNodeRef, CATEGORY_NAME);
+        then(categoryServiceMock).shouldHaveNoMoreInteractions();
```

Comment:
I am in favour of covering as much as possible in a unit test.
Once we change the implementation and add a call to some mocked method this can be a fail fast way to find what needs to be mocked.